chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -155,23 +164,31 @@ impl<'a> TransactionSourceForConnect<'a> {
 pub struct TransactionVerifierDelta {
     tx_index_cache: BTreeMap<OutPointSourceId, CachedInputsOperation>,
     utxo_cache: ConsumedUtxoCache,
-    utxo_block_undo: BTreeMap<TransactionSource, BlockUndoEntry>,
+    utxo_block_undo: BTreeMap<TransactionSource, UtxosBlockUndoEntry>,
     token_issuance_cache: ConsumedTokenIssuanceCache,
+    accounting_delta: PoSAccountingDeltaData,
+    accounting_delta_undo: BTreeMap<TransactionSource, AccountingBlockUndoEntry>,
 }
 
 /// The tool used to verify transaction and cache their updated states in memory
-pub struct TransactionVerifier<C, S, U> {
+pub struct TransactionVerifier<C, S, U, A> {
     chain_config: C,
     storage: S,
+    best_block: Id<GenBlock>,
+
     tx_index_cache: OptionalTxIndexCache,
-    utxo_cache: UtxosCache<U>,
-    utxo_block_undo: BTreeMap<TransactionSource, BlockUndoEntry>,
     token_issuance_cache: TokenIssuanceCache,
-    best_block: Id<GenBlock>,
+
+    utxo_cache: UtxosCache<U>,
+    utxo_block_undo: UtxosBlockUndoCache,
+
+    accounting_delta: PoSAccountingDelta<A>,
+    accounting_delta_undo: AccountingBlockUndoCache,
```

Comment:
I would call this `accounting_block_undo`, just for symmetry, since they both serve the same purpose from different angles.

I understand that the "block" terminology might be wrong, in which case I also would seek a better name than `utxo_block_undo` as well.

Not a priority, but maybe give it a thought.